From 7f8f5e1f4cadf7f7693fe914a3593abf4f66d258 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Tue, 13 Mar 2012 15:09:24 +0100
Subject: [PATCH 1/3] cpu flags: aliases: pclmuldq|pclmulqdq and
 ffxsr|fxsr_opt

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1331651366-9586-2-git-send-email-ehabkost@redhat.com>
Patchwork-id: 38486
O-Subject: [RHEL6 qemu-kvm PATCH 1/3] cpu flags: aliases: pclmuldq|pclmulqdq and ffxsr|fxsr_opt
Bugzilla: 767302
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>

Bugzilla: 767302
Upstream: applied
(cherry picked from commit f370be3c50c0aec7785e0f82c35a0e953f23dc5b)

pclmulqdq: /proc/cpuinfo on Linux and all documentation I have seen uses
pclmulqdq as the flag name. As the only document using pclmuldq seems to
be the Intel CPUID documentation (Application Note 485), it looks like a
typo and not the correct name for the flag.

ffxsr: AMD docs refer to fxsr_opt as ffxsr, so allow this named to be
used too.

Cc: mkletzan@redhat.com
Cc: armbru@redhat.com
Cc: alex.williamson@redhat.com
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
---
 target-i386/cpuid.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 target-i386/cpuid.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/target-i386/cpuid.c b/target-i386/cpuid.c
index 146dde3..70f9c93 100644
--- a/target-i386/cpuid.c
+++ b/target-i386/cpuid.c
@@ -42,7 +42,7 @@ static const char *feature_name[] = {
     "ht" /* Intel htt */, "tm", "ia64", "pbe",
 };
 static const char *ext_feature_name[] = {
-    "pni|sse3" /* Intel,AMD sse3 */, "pclmuldq", NULL, "monitor",
+    "pni|sse3" /* Intel,AMD sse3 */, "pclmulqdq|pclmuldq", NULL, "monitor",
     "ds_cpl", "vmx", NULL /* Linux smx */, "est",
     "tm2", "ssse3", "cid", NULL,
     NULL, "cx16", "xtpr", NULL,
@@ -58,7 +58,7 @@ static const char *ext2_feature_name[] = {
     "mtrr", "pge", "mca", "cmov",
     "pat", "pse36", NULL, NULL /* Linux mp */,
     "nx|xd", NULL, "mmxext", "mmx",
-    "fxsr", "fxsr_opt" /* AMD ffxsr */, "pdpe1gb" /* AMD Page1GB */, "rdtscp",
+    "fxsr", "fxsr_opt|ffxsr" /* AMD ffxsr */, "pdpe1gb" /* AMD Page1GB */, "rdtscp",
     NULL, "lm|i64", "3dnowext", "3dnow",
 };
 static const char *ext3_feature_name[] = {
-- 
1.7.7.6

